packages/form-js-viewer/src/render/components/form-fields/Number.js

```diff
@@ -11,31 +13,137 @@ import {
   prefixId
 } from '../Util';
 
+import {
+  isNullEquivalentValue,
+  isValidNumber,
+  willKeyProduceValidNumber
+} from '../util/numberFieldUtil';
+
 const type = 'number';
 
 
-export default function Number(props) {
+export default function Numberfield(props) {
   const {
     disabled,
     errors = [],
     field,
-    value
+    value,
+    onChange
   } = props;
 
   const {
     description,
     id,
     label,
-    validate = {}
+    validate = {},
+    decimalDigits,
+    serializeToString = false,
+    increment
   } = field;
 
   const { required } = validate;
 
-  const onChange = ({ target }) => {
-    props.onChange({
-      field,
-      value: Number.sanitizeValue({ value: target.value })
-    });
+  const inputRef = useRef();
+
+  const [ stringValueCache, setStringValueCache ] = useState('');
+
+  const valueCacheMismatch = useMemo(() => Numberfield.sanitizeValue({ value, formField: field }) !== Numberfield.sanitizeValue({ value: stringValueCache, formField: field }), [ stringValueCache, value, field ]);
```

Comment:
This is quite a statement, I think we should add some comments here to give the least amount of context.

Reply:
Wrote some good comments here